Services/OpenIdConnect/classes/class.ilAuthProviderOpenIdConnect.php

```diff
@@ -41,16 +42,21 @@ public function handleLogout()
             return false;
         }
 
-        $auth_token = ilSession::get('oidc_auth_token');
-        $this->getLogger()->debug('Using token: ' . $auth_token);
+        $id_token = ilSession::get(self::OIDC_AUTH_IDTOKEN);
+        $this->getLogger()->debug('Logging out with token: ' . $id_token);
 
-        if (strlen($auth_token)) {
-            ilSession::set('oidc_auth_token', '');
+
+        if (strlen($id_token)) {
```

Comment:
We are reading this from the session storage, so it might theoretically be `null`.

```suggestion
        if (is_string($id_token) && $id_token !== '') {
```